sources/api/migration/migrations/v1.4.0/registry-mirror-representation/src/main.rs

```diff
@@ -0,0 +1,126 @@
+#![deny(rust_2018_idioms)]
+
+use datastore::{Key, KEY_SEPARATOR};
+use migration_helpers::{migrate, Migration, MigrationData, Result};
+use serde_json::{Map, Value};
+use std::collections::HashMap;
+use std::process;
+
+const MIRRORS_SETTING_NAME: &'static str = "settings.container-registry.mirrors";
+
+/// This migration changes the model type of `settings.container-registry.mirrors` from `HashMap<SingleLineString, Vec<Url>>`
+/// to `Vec<RegistryMirrors>` on upgrade and vice-versa on downgrades.
+pub struct ChangeRegistryMirrorsType;
+
+impl Migration for ChangeRegistryMirrorsType {
+    /// Newer versions store `settings.container-registry.mirrors` as `Vec<RegistryMirrors>`.
+    /// Need to convert from `HashMap<SingleLineString, Vec<Url>>`.
+    fn forward(&mut self, mut input: MigrationData) -> Result<MigrationData> {
+        let mirrors: HashMap<_, _> = input
+            .data
+            .iter()
+            .filter(|&(k, _)| k.starts_with(format!("{}.", MIRRORS_SETTING_NAME).as_str()))
+            .map(|(k, v)| (k.to_owned(), v.to_owned()))
+            .collect();
+        let mut new_mirrors = Vec::new();
+        for (setting, endpoint) in mirrors {
+            // Need to split the full setting name to get the registry name.
+            // We get the index of the third '.' and split the string around there and leave out the '.'.
+            // Trim quotation marks around the registry name.
+            let registry = setting
```

Comment:
I think this block can use `strip_prefix` instead to simplify?  (Probably still need to trim quotes, otherwise migrating forward and backward repeatedly would add quotes forever, I think.)